### PR TITLE
Introducing acceptance test

### PR DESCRIPTION
**Pull request by RoyUP9**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: tests
```
Comment (IgorGov) [inline .github/workflows/tests.yml:1]: lets call it "acceptance tests"

File: Makefile [hunk 2]
```diff
@@ -65,8 +71,11 @@ clean-cli:  ## Clean CLI.
 clean-docker:
 	@(echo "DOCKER cleanup - NOT IMPLEMENTED YET " )
 
-test-cli: ## Run tests.
+test-cli: ## Run cli tests.
```
Comment (IgorGov) [inline Makefile:74]: it feels like the comment redundant here, we have the same text in echo...

File: Makefile [hunk 3]
```diff
@@ -65,8 +71,11 @@ clean-cli:  ## Clean CLI.
...
 	@echo "running cli tests"; cd cli && $(MAKE) test
 
-test-agent: ## Run tests.
+test-agent: ## Run agent tests.
```
Comment (IgorGov) [inline Makefile:77]: it feels like the comment redundant here, we have the same text in echo...

File: Makefile [hunk 4]
```diff
@@ -65,8 +71,11 @@ clean-cli:  ## Clean CLI.
...
 	@echo "running agent tests"; cd agent && $(MAKE) test
+
+system-test: ## Run system tests.
```
Comment (IgorGov) [inline Makefile:80]: it feels like the comment redundant here, we have the same text in echo...

Comment (IgorGov) [reply Makefile:80]: lets call it acceptance tests

File: build-push-featurebranch.sh [hunk 5]
```diff
@@ -1,15 +1,22 @@
...
 set -e
 
-SERVER_NAME=mizu
 GCP_PROJECT=up9-docker-hub
 REPOSITORY=gcr.io/$GCP_PROJECT
-GIT_BRANCH=$(git branch | grep \* | cut -d ' ' -f2 | tr '[:upper:]' '[:lower:]')
-SEM_VER=${SEM_VER=0.0.0}
+SERVER_NAME=mizu
+
+if [ "$CI" != 'true' ]
```
Comment (IgorGov) [inline build-push-featurebranch.sh:8]: can separate this file to multiple files, and use it without additional flags (like $CI).
For instance a script which builds the image, a script which push the image, a script which builds and push.
in make docker run only the build img script...

File: cli/config/configStruct.go [hunk 6]
```diff
@@ -16,6 +17,7 @@ type ConfigStruct struct {
 	Version                configStructs.VersionConfig `yaml:"version"`
 	View                   configStructs.ViewConfig    `yaml:"view"`
 	AgentImage             string                      `yaml:"agent-image,omitempty" readonly:""`
+	ImagePullPolicyStr     string                      `yaml:"image-pull-policy" default:"Always"`
```
Comment (IgorGov) [inline cli/config/configStruct.go:20]: should it be "readonly"?

Comment (RoyUP9) [reply cli/config/configStruct.go:20]: i dont think it should be, then we won't be able to build and run local images

File: test-installations.sh [hunk 7]
```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
```
Comment (IgorGov) [inline test-installations.sh:1]: lets move it to tests folder

File: build-agent-ci.sh [hunk 8]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
```
Comment (IgorGov) [inline build-agent-ci.sh:1]: can we move it to agent folder

File: acceptanceTests/testsUtils.go [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+	dir, filePathErr := os.Getwd()
+	if filePathErr != nil {
+		return "", filePathErr
+	}
+
+	cliPath := path.Join(dir, "../cli/bin/mizu_ci")
+	return cliPath, nil
+}
+
+func GetDefaultCommandArgs() []string {
```
Comment (IgorGov) [inline acceptanceTests/testsUtils.go:25]: lets make it private to package

File: acceptanceTests/tap_test.go [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+					t.Logf("failed to cleanup tap command, err: %v", err)
+				}
+			})
+
+			if err := tapCmd.Start(); err != nil {
+				t.Errorf("failed to start tap command, err: %v", err)
+				return
+			}
+
+			time.Sleep(30 * time.Second)
```
Comment (IgorGov) [inline acceptanceTests/tap_test.go:41]: lets change it to wait loop with timeout for "get tapped pods"

File: acceptanceTests/tap_test.go [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+
+			proxyUrl := "http://localhost:8080/api/v1/namespaces/mizu-tests/services/httpbin/proxy/get"
+			for i := 0; i < entriesCount; i++ {
+				if _, requestErr := ExecuteHttpRequest(proxyUrl); requestErr != nil {
+					t.Errorf("failed to send proxy request, err: %v", requestErr)
+					return
+				}
+			}
+
+			time.Sleep(5 * time.Second)
```
Comment (IgorGov) [inline acceptanceTests/tap_test.go:51]: lets replace with loop with timeout

**Review by IgorGov — APPROVED** [review]

**Pull request merged**